Sources/TuistGenerator/Generator/ProjectGenerator.swift

```diff
@@ -23,9 +23,16 @@ extension ProjectConstants {
 }
 
 protocol ProjectGenerating: AnyObject {
+    /// Generates the given project.
+    /// - Parameters:
+    ///   - project: Project to be generated.
+    ///   - graph: Dependencies graph.
+    ///   - sourceRootPath: Directory where the files are relative to.
+    ///   - xcodeprojPath: Path to the Xcode project. When not given, the xcodeproj is generated at sourceRootPath.
     func generate(project: Project,
                   graph: Graphing,
-                  sourceRootPath: AbsolutePath?) throws -> GeneratedProject
+                  sourceRootPath: AbsolutePath?,
```

Comment:
Maybe it should also have default parameters? In other places it seems to have them

Reply:
Why do we have default parameters at all in the other cases?

Reply:
So this allows configuring 2 things:
- What we configure as a source root.
- Where the Xcode project is generated.

When any of the arguments is nil we assume the path is the directory that contains the manifest. I'll update this one to be consistent with the others.

Reply:
Just noticed that this is a protocol and therefore we can't define default arguments.